### DISCUSSION TITLE
Invalid model: Using Bytes type is not allowed in the model's id

### DISCUSSION THREAD

**Discussion by mwstr** [prisma/prisma#10399] [Q&A]
Provider: sqlserver.
I have no permission to change columns types. Some tables have ids of `@db.Binary(16)` type. But it's not allowed in prisma. Why? And is there any ways to work around this problem?

**Comment by austincrim** [comment]
Hey @MeowningMaster 👋!

This should be allowed in the latest version of the Prisma Client. Please update and regenerate the client and you should be good to go!

Let me know if you have any issues.

**Reply by aelx311** [reply]
I have 3.6.0 but still got the error when running npx prisma generate.

Error: Schema parsing
error: Invalid model: Using Bytes type is not allowed in the model's id..
  -->  schema.prisma:5615
model tblExample {
ColumnA String @db.VarChar(128)
ColumnB Bytes  @db.VarBinary(128)
ColumnC String @db.VarChar(128)

@@id([ColumnA , ColumnB], map: "PK_tblExample")
@@unique([ColumnA , ColumnC ], map: "AK_tblExample")
}

**Reply by nicolashenry** [reply]
@austincrim Is it supposed to work with the latest version of the Prisma Client?

**Reply by austincrim** [reply]
@nicolashenry Yes, it should be working. Are you seeing something different? What version are you using?

**Reply by nicolashenry** [reply]
@austincrim 
Using Prisma 4.5.0 and this model (extracted from a SQL Server database with `prisma db pull`):
![image](https://user-images.githubusercontent.com/139160/197815291-d4ef6260-cc0f-49ef-8f55-601eef3669a5.png)
`prisma generate` give me this:
![image](https://user-images.githubusercontent.com/139160/197815588-d0c4747b-86cf-4cbd-8f5f-f1e1d731df59.png)

**Reply by austincrim** [reply]
Ah, it looks like this is just an issue when using SQL Server. I'm chatting with our engineers now to determine if this is a bug or intentional behavior.

**Reply by austincrim** [reply]
I confirmed this is a bug we will fix soon. You can watch for updates [here](https://github.com/prisma/prisma/issues/15986).

**Reply by nicolashenry** [reply]
@austincrim Thanks a lot!

**Comment by mwstr** [comment]
Sorry, I already realized that Prisma does not suit me, not only for this reason. And I don't have time to return everything to make a test

**Reply by austincrim** [reply]
@MeowningMaster sorry to hear that!

Do you have any specific feedback that I could take back to our product and engineering teams?

**Reply by mwstr** [reply]
No, I do not

**Comment by AbianS** [comment]
Is there still no solution to this problem?

**Reply by janpio** [reply]
The bug is still open here: https://github.com/prisma/prisma/issues/15986

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
